src/bin/stratisd.rs

```diff
@@ -109,8 +133,13 @@ fn run() -> StratisResult<()> {
             }
         }
 
-        // Ask the engine to check its pools
-        engine.borrow_mut().check()
+        // Refresh list of dbus fds to poll for every time. This can change as
+        // D-Bus clients come and go.
+        fds.truncate(dbus_fd_start_idx);
+        fds.extend(dbus_conn.watch_fds().iter().map(|w| w.to_pollfd()));
```

Comment:
In this new arrangement, the first time through the loop there are no d-bus events to be interested in, but there were before. Is that intentional?

Reply:
The loop is structured to avoid duplicating code. There are things that need to happen before `poll()` is called the first time (setup), and then there are things (checking, setup) every time `poll()` wakes up. That ends up putting the call to `poll()` at the bottom of the loop, so the first time through the "checking" part is moot, we're just doing the "setup" part... and then in subsequent loops both checking and setup are done.